### PR TITLE
Refactor how file cache entries are stored

### PR DESCRIPTION
**Pull request by KoBeWi**
Currently the code for writing FileSystem cache is basically a one unreadable line of string additions, followed by a loop of string additions. This PR changes it to be much more readable, without affecting the file's content. It might also make the code faster.

### CODE REVIEW

**Review by akien-mga — APPROVED** [review]

File: editor/editor_file_system.cpp [hunk 1]
```diff
@@ -1333,24 +1333,28 @@ void EditorFileSystem::_save_filesystem_cache(EditorFileSystemDirectory *p_dir,
...
-			s += p_dir->files[i]->deps[j];
+		String type = file_info->type;
+		if (file_info->resource_script_class) {
+			type += "/" + String(file_info->resource_script_class);
 		}
 
-		p_file->store_line(s);
+		PackedStringArray cache_string;
+		cache_string.append(file_info->file);
+		cache_string.append(file_info->type);
```
Comment (bruvzg) [inline editor/editor_file_system.cpp:1348]: Should it be it `file_info->type` or `type` modified earlier?

Comment (KoBeWi) [reply editor/editor_file_system.cpp:1348]: Ah it should be `type`.

File: editor/editor_file_system.cpp [hunk 2]
```diff
@@ -1333,24 +1333,28 @@ void EditorFileSystem::_save_filesystem_cache(EditorFileSystemDirectory *p_dir,
...
-				s += "<>";
-			}
-			s += p_dir->files[i]->deps[j];
+		String type = file_info->type;
+		if (file_info->resource_script_class) {
+			type += "/" + String(file_info->resource_script_class);
 		}
 
-		p_file->store_line(s);
+		PackedStringArray cache_string;
```
Comment (Mickeon) [inline editor/editor_file_system.cpp:1346]: StringBuilder isn't appropriate for this, is it? Code is sooo much more readable now, but just putting the thought out there.

Comment (KoBeWi) [reply editor/editor_file_system.cpp:1346]: I used PackedStringArray for easier joining.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**